app/models/post.rb

```diff
@@ -1,17 +1,21 @@
-class Post < ActiveRecord::Base
-  belongs_to :user, class_name: 'User', foreign_key: :user_id
-  has_many :likes
-  has_many :comments
+class Post < ApplicationRecord
+  validates :title, presence: true, length: { maximum: 250, too_long: '%<count>s characters is the maximum allowed' }
+  validates :comments_counter, :likes_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
 
-  def five_last_comments
-    comments.order('created_at Desc').limit(5)
-  end
+  belongs_to :user, counter_cache: :posts_counter
+  has_many :comments, dependent: :destroy
+  has_many :likes, counter_cache: :likes_counter
+  has_many :users, through: :likes, counter_cache: :likes_counter
 
-  after_save :update_posts_counter
+  def recent_comments(limit = 5)
+    comments.last(limit)
+  end
 
-  private
+  def liked?(user_id)
+    likes.where(user_id:).exists?
```

Comment:
- [ ] This issue has not been resolved because I still have the same error pointed out by the previous code reviewer. I suggest you rename the parameter passed to the liked method so that you can resolve the error and also pass the linter check. Also please kindly check why you pushed more than 5000 files and ensure you ignore the files that are supposed to be git ignored. It is really hard to access the files changed with more than 5000 files changed.


```suggestion
  def liked?(the_user_id)
    likes.where(user_id: the_user_id).exists?
```

Reply:
This issue has been resolved